appveyor.yml

```diff
@@ -121,8 +121,13 @@ build_script:
   - "%CMD_IN_ENV% python setup.py build"
 
 test_script:
-  # Run the project tests
-  - "%CMD_IN_ENV% python setup.py nosetests"
+  # Run the project tests and upload results
+  - ps: |
+      &$env:PYTHON\python setup.py nosetests --with-xunit
+      if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
+      # upload results to AppVeyor
```

Comment:
This should be in `on_success:` & `on_failure:` , or in `after_test:`

Reply:
`on_finish` - https://help.appveyor.com/discussions/questions/1479-after_test-not-invoked-when-test-fails